STY: update with use of plt.subplots(), other readability edits

mscross opened the pull request:
* Edit example to interface via `fig`, `ax0, ax1` rather than `plt`

* Adjust vertical spacing between subplots to fix title, tick label overlap

![image](https://cloud.githubusercontent.com/assets/6813152/7235534/5dde8d50-e754-11e4-9a54-dcab576357fc.png)


* Remove whitespace around some operators to improve readability

tacaswell commented:
This broke something in building the examples https://travis-ci.org/matplotlib/matplotlib/jobs/59272525

jenshnielsen commented:
ax0.title('pcolormesh with levels') etc should be ax0.set_title('pcolormesh with levels')

jenshnielsen commented:
In general I would prefer fig.tight_layout() over fig.subplots_adjust(hspace=0.35) when possible. Can you try that to see if it works as intended?

mscross commented:
Fixed `ax0.set_title()` (same for ax1), used `fig.tight_layout()`, which gets the desired results.

Also fixed a branch issue that accidentally brought in commits from a previous unrelated PR (that broke the travis build).

tacaswell commented:
@mscross Thank you!
